cmd/controller/main.go

```diff
@@ -74,5 +75,6 @@ func main() {
 	sharedmain.MainWithContext(injection.WithNamespaceScope(signals.NewContext(), *namespace), ControllerLogKey,
 		taskrun.NewController(*namespace, images),
 		pipelinerun.NewController(*namespace, images),
+		run.NewController,
```

Comment:
sorry coming late to the party 😢 but staring at this `run` here looks incomplete, is it possible to rename it to something descriptive? 😜

Reply:
The `run` package is generated based on the name of the type, which is just `Run`. I can't really think of a better name for it, but ideas are welcome. An earlier iteration of the design called it `CustomTaskRun` but that got to be a mouthful/keyboardful...

Reply:
is it distinguishable as a `Task` or a `Pipeline`?  How about `CustomRun`? 😜

Reply:
The general idea is that a `Run` reports the execution of a generic...something. What that is is up to the custom task author -- it could be running a remote build process, "running" a wait between two other Tasks, "running" an approval.

It's sort of like a generic container to express an ongoing execution of a thing. In the future we could experiment with having `Run<Task>` and `Run<Pipeline>` in addition to (eventually replacing?) `TaskRun` and `PipelineRun`. This would give us flexibility to experiment with new definitions of `Task` and `Pipeline` outside the `tektoncd/pipeline` repo, or to mix versions of these types (e.g., `Run<v1alpha1.Task>` alongside `Run<v1.Task>` in the same `PipelineRun`).

So I think even `CustomRun` is still a bit wordy for what it could end up supporting. What's a "custom" run in a possible future world where that's actually the _only_ kind of running thing? I personally actually kind of like the succintness of "`Run`"  in that context. 🤷‍♂️

Reply:
cool, sounds good, thanks 👍